src/samples/Authorization/ManualSignIn/AuthAgent.cs

```diff
@@ -0,0 +1,224 @@
+    }
+
+    /// <summary>
+    /// This method is called by the `OnMessage("-signin", SignInAsync);` registration to handle the sign-in process for the user. 
+    /// This is a generic message handler pattern used by the agent SDK to allow a developer to attach specific implementation to a specific message.
+    /// </summary>
+    /// <param name="turnContext"><see cref="ITurnContext"/></param>
+    /// <param name="turnState"><see cref="ITurnState"/></param>
+    /// <param name="cancellationToken"><see cref="CancellationToken"/></param>
+    private async Task SignInAsync(ITurnContext turnContext, ITurnState turnState, CancellationToken cancellationToken)
+    {
+
+        // We are using the Authorization class to sign in the user to the authorization provider.
+        // This triggers the User login flow, and the user will be prompted to sign in using the authorization provider if needed.
+        // On Success, this will call the OnUserSignInSuccess event handler to notify the user of the success and provide the access token.  The token will also be available in the Authorization class.
+
+        await UserAuthorization.SignInUserAsync(turnContext, turnState, _signInHandlerName, cancellationToken: cancellationToken);
+    }
+
+    /// <summary>
+    /// Raised when the user successfully signs in to the authorization provider.
+    /// This also provides the access token for the user.
+    /// </summary>
+    /// <param name="turnContext"><see cref="ITurnContext"/></param>
+    /// <param name="turnState"><see cref="ITurnState"/></param>
+    /// <param name="cancellationToken"><see cref="CancellationToken"/></param>
+    /// <param name="handlerName">this is the authorization handler that was used to execute the login.</param>
+    /// <param name="token">This is the access token for the user</param>
+    /// <param name="initiatingActivity">This is the activity that started the login flow, note that it may be different then the activity on the turnContext. </param>
+    /// <returns></returns>
+    private async Task OnUserSignInSuccess(ITurnContext turnContext, ITurnState turnState, string handlerName, string token, IActivity initiatingActivity, CancellationToken cancellationToken)
+    {
+        // The user has successfully signed in to the authorization provider and we will continue the turn. 
+
+        // Handle the request for the designated login handler. 
+        if (handlerName.Equals(_signInHandlerName))
+        {
+            // Check for Access Token from the Authorization Sub System. 
+            if (string.IsNullOrEmpty(UserAuthorization.GetTurnToken(handlerName)))
+            {
+                // Failed to get access token here, and we will now bail out of this message loop. 
+                await turnContext.SendActivityAsync($"The auto sign in process failed and no access token is available", cancellationToken: cancellationToken);
```

Comment:
@Mattb-msft is this correct? 

Seems weird that the OnSignInSuccess event can lead into a failed signin

ni: this is the manual samples, not sure why the message says auto.